.github/workflows/unittests.yml

```diff
@@ -125,8 +125,19 @@ jobs:
           sudo apt install libharfbuzz-dev libfribidi-dev libgit2-dev libcurl4-openssl-dev
         shell: bash
 
+      - name: Set environment variable to never compile packages on Windows and macOS
+        if: runner.os != 'Linux' && matrix.r != 'devel'
+        run: |
+          # don't try to install new pkgs versions from source (which often requires additional system dependencies which, when missing, lead to errors, e.g., ragg).
+          cat("\noptions(install.packages.compile.from.source = \"never\")\n", file = "~/.Rprofile", sep = "", append = TRUE)
+        shell: Rscript {0}
+        
       - uses: r-lib/actions/setup-r-dependencies@v2
         with:
+          extra-packages: |
+            any::rcmdcheck
+            github::jasp-stats/jaspTools
+            cran::igraph
```

Comment:
why do you always install `igraph`?

Reply:
ah, because I learn that we want to install from CRAN so  specify it by `cran::igraph` , but I overlook the needs of `if: runner.os == 'Linux' && inputs.needs_igraph` , sorry for that and I'll give it back.

Reply:
The reason for this is that the Linux binary of igraph that RStudio has (the RSPM repository) doesn't work properly. If this is no longer necessary (e.g., r-lib fixes this somehow) it would be nice if we could do without that step. But we can also leave that for another time of course :)

Reply:
Yes, binary packages are now preferred from CRAN unless we specify the source, the reason I rollback it is that `igraph` is only required in some modules.